### DISCUSSION TITLE
no data

### DISCUSSION THREAD

**Discussion by rcr00056** [teslamate-org/teslamate#2517] [Getting started]
I have logged into the :4000 Teslamate and it shows my car, says its charging, etc. all looks good.

when I go to the grafana dash it all says no data. no battery level or anything. Ive updated docker compose. I’ve updated docker, rebooted my pi. I don’t know how to gather a log if that would be helpful. 

Any help please. Thanks

**Comment by cwanja** [comment]
What do the Grafana logs say? What is the data source for Grafana?

**Reply by rcr00056** [reply]
How do I see grafana logs?
edit: I think I found out how... docker-compose logs via terminal.

Im getting tons of errors in grafana like " lvl=eror msg="Metric request error" logger=context userId=1 orgId=1 uname=admin error="failed to query data: invalid option: \"mypassword\"

**Reply by cwanja** [reply]
It looks like your database setup is not correct. What is in your YAML file? Did you copy and paste from the docs? https://docs.teslamate.org/docs/installation/docker

**Reply by rcr00056** [reply]
thank you again for the responses. I fixed it! sorry I'm in healthcare and do not know how to code...
I had a SPACE before the  password and that screwed it all up.

you will probably laugh but I didn't know how to edit or erase the yml file so I just started all over fresh and installed everything from new.

**Reply by cwanja** [reply]
Hah no worries at all! Glad you got it fixed and squared away. Enjoy TeslaMate and your Tesla!

**Reply by mikerintz** [reply]
Thanks for this! I made the same mistake with the extra space before the password. I DO know how to edit the YAML file, but I still had to wipe the docker volume and set it up from scratch because just changing the password in the yml file wasn't enough. The postgres database was provisioned with the corrupted password during the initial build, and I didn't know how to fix that.

**Reply by albertcwong** [reply]
happened to me, looks like the default yaml file has a space.  I diagnose, fix the password and it's all good. Then power outage, and I do this all over again. It's getting old. What's the permanent fix?

**Reply by albertcwong** [reply]
If I restart the grafana container, the password needs to be reset on the UI. How do I get the updated password without the space to stick?

**Reply by albertcwong** [reply]
The environment variable on the container also stores the wrong password, so changing that fixes the password issue. I see this is not an uncommon issue after searching around, so it might be worth updating the yaml file with the spaces removed after the '=' or figuring out why the password inconsistently propagates with or without the password depending on the application
